CppSamples/Layers/AddBuildingSceneLayer/AddBuildingSceneLayer.cpp

```diff
@@ -0,0 +1,134 @@
+// [WriteFile Name=AddBuildingSceneLayer, Category=Layers]
+// [Legal]
+// Copyright 2025 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#ifdef PCH_BUILD
+#include "pch.hpp"
+#endif // PCH_BUILD
+
+// sample headers
+#include "AddBuildingSceneLayer.h"
+
+// ArcGIS Maps SDK headers
+#include "ArcGISTiledElevationSource.h"
+#include "BuildingSceneLayer.h"
+#include "BuildingSublayer.h"
+#include "Camera.h"
+#include "ElevationSourceListModel.h"
+#include "Error.h"
+#include "LayerListModel.h"
+#include "LocalSceneQuickView.h"
+#include "MapTypes.h"
+#include "Point.h"
+#include "Scene.h"
+#include "SceneViewTypes.h"
+#include "SpatialReference.h"
+#include "Surface.h"
+#include "Viewpoint.h"
+
+using namespace Esri::ArcGISRuntime;
+
+AddBuildingSceneLayer::AddBuildingSceneLayer(QObject* parent /* = nullptr */):
+  QObject(parent),
+  m_scene(new Scene(BasemapStyle::ArcGISTopographic, SceneViewingMode::Local, this))
+{
+  // create a new elevation source from Terrain3D REST service
+  ArcGISTiledElevationSource* elevationSource = new ArcGISTiledElevationSource(
+    QUrl("https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer"), this);
+
+  // add the elevation source to the scene to display elevation
+  m_scene->baseSurface()->elevationSources()->append(elevationSource);
+}
+
+AddBuildingSceneLayer::~AddBuildingSceneLayer() = default;
+
+void AddBuildingSceneLayer::init()
+{
+  // Register classes for QML
+  qmlRegisterType<LocalSceneQuickView>("Esri.Samples", 1, 0, "LocalSceneView");
+  qmlRegisterType<AddBuildingSceneLayer>("Esri.Samples", 1, 0, "AddBuildingSceneLayerSample");
+}
+
+LocalSceneQuickView* AddBuildingSceneLayer::localSceneView() const
+{
+  return m_localSceneView;
+}
+
+// Set the view (created in QML)
+void AddBuildingSceneLayer::setLocalSceneView(LocalSceneQuickView* localSceneView)
+{
+  m_layerLoaded = false;
```

Comment:
You can simplify this bool down to `m_buildingSceneLayer->loadStatus() == LoadStatus::Loaded`. Do you need to store it separately?